mantidproject/mantid
# Review IndirectCommon file

### Description of work
This PR does some refactoring of the IndirectCommon.py file:
- It removes unused functions
- It moves one or two functions to different files if they are only used in one location
- It renames functions to use snake case
- It uses python typing for function parameter and return types
- It removes uses of mtd["name"] in preference of AnalysisDataService.retrieve
- It uses python f strings rather than %s notation

The PR also fixes a bug in the CheckForSampleLogs algorithm where it was ignoring the `EnableLogging=False` flag because it was not using the global log held by the algorithm object.

Fixes #37773 
Fixes #37776

### To test:
The IndirectCommonTests suite should pass Ok.

1. Open Inelastic -> Data Processor interface
1.  Load sample 261176 _red
1. Load resolution 26173 _res
1. Click Run
1. See that the logger does NOT have the following message
```
Property deltaE-mode not found
```

Also run through the testing instructions for Indirect Data Reduction:
https://developer.mantidproject.org/Testing/Indirect/DataReductionTests.html


### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
